From b6401f66a93e9f0d19ed2f26e320b9ac4b5ca490 Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Tue, 5 Jan 2016 15:20:34 +0200
Subject: [PATCH 0648/1240] nfc: Adapt pxa3xx_nand driver to compile after
 driver cherry-pick

	- Fix compilation issues.
	- Add required macros for register base adddress.
	- Some of the addresses and values are hardcoded for now, will be
	  changed with DT based configuration in a subsequnt patch.

Change-Id: Ifd0fd9f1f04dca13193ba3922fd4ac29396dd26e
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/26569
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm/cpu/armv7/armada38x/clock.c            |  8 ++++++++
 arch/arm/cpu/armv8/armada8k/clock.c             |  8 ++++++++
 arch/arm/include/asm/arch-armada38x/regs-base.h |  2 ++
 arch/arm/include/asm/arch-armada8k/regs-base.h  |  2 ++
 arch/arm/include/asm/arch-mvebu/clock.h         |  4 ++++
 board/mvebu/common/Makefile                     |  1 -
 board/mvebu/common/nand_init.c                  | 27 -------------------------
 configs/mvebu_armada38x_defconfig               | 12 +++++------
 drivers/mtd/nand/pxa3xx_nand.c                  |  2 +-
 include/configs/mvebu-common.h                  |  3 ++-
 10 files changed, 33 insertions(+), 36 deletions(-)
 delete mode 100644 board/mvebu/common/nand_init.c

diff --git a/arch/arm/cpu/armv7/armada38x/clock.c b/arch/arm/cpu/armv7/armada38x/clock.c
index e69c204..e5c4440 100644
--- a/arch/arm/cpu/armv7/armada38x/clock.c
+++ b/arch/arm/cpu/armv7/armada38x/clock.c
@@ -47,6 +47,14 @@ u32 soc_timer_clk_get(void)
 	return 800000000;
 }
 
+
+#ifdef CONFIG_NAND_PXA3XX
+unsigned long mvebu_get_nand_clock(void)
+{
+	return 0;
+}
+#endif
+
 u32 get_fdt_tclk(const void *blob, int node)
 {
 	u32 tclk;
diff --git a/arch/arm/cpu/armv8/armada8k/clock.c b/arch/arm/cpu/armv8/armada8k/clock.c
index bce2082..e2d6265 100644
--- a/arch/arm/cpu/armv8/armada8k/clock.c
+++ b/arch/arm/cpu/armv8/armada8k/clock.c
@@ -60,6 +60,14 @@ u32 soc_tclk_get(void)
 #endif
 }
 
+
+#ifdef CONFIG_NAND_PXA3XX
+unsigned long mvebu_get_nand_clock(void)
+{
+	return 0;
+}
+#endif
+
 u32 soc_get_cpu_clk(void)
 {
 	u32 pll_idx = (readl(SAMPLE_AT_RESET_REG_1) & SAR1_PLL2_MASK) >> SAR1_PLL2_OFFSET;
diff --git a/arch/arm/include/asm/arch-armada38x/regs-base.h b/arch/arm/include/asm/arch-armada38x/regs-base.h
index 682765f..44c47c1 100644
--- a/arch/arm/include/asm/arch-armada38x/regs-base.h
+++ b/arch/arm/include/asm/arch-armada38x/regs-base.h
@@ -36,4 +36,6 @@
 
 #define MVEBU_IO_ERR_CTL_REG	(MVEBU_REGS_BASE + 0x20200)
 
+#define MVEBU_NAND_BASE		(MVEBU_REGS_BASE + 0xD0000)
+
 #endif	/* _REGS_BASE_H_ */
diff --git a/arch/arm/include/asm/arch-armada8k/regs-base.h b/arch/arm/include/asm/arch-armada8k/regs-base.h
index 6b5c9e7..a469c4a 100644
--- a/arch/arm/include/asm/arch-armada8k/regs-base.h
+++ b/arch/arm/include/asm/arch-armada8k/regs-base.h
@@ -43,4 +43,6 @@
 
 #define MVEBU_LLC_BASE		(MVEBU_REGS_BASE + 0x8000)
 
+#define MVEBU_NAND_BASE		(MVEBU_REGS_BASE + 0x720000)
+
 #endif	/* _REGS_BASE_H_ */
diff --git a/arch/arm/include/asm/arch-mvebu/clock.h b/arch/arm/include/asm/arch-mvebu/clock.h
index 7a79f10..9b2926d 100644
--- a/arch/arm/include/asm/arch-mvebu/clock.h
+++ b/arch/arm/include/asm/arch-mvebu/clock.h
@@ -29,6 +29,10 @@ u32 soc_cpu_clk_get(void);
 u32 soc_ddr_clk_get(void);
 u32 soc_timer_clk_get(void);
 
+#ifdef CONFIG_NAND_PXA3XX
+unsigned long mvebu_get_nand_clock(void);
+#endif
+
 u32 soc_clock_get(const void *blob, int node);
 u32 get_fdt_tclk(const void *blob, int node);
 void soc_print_clock_info(void);
diff --git a/board/mvebu/common/Makefile b/board/mvebu/common/Makefile
index 0405070..7dbdd5c 100644
--- a/board/mvebu/common/Makefile
+++ b/board/mvebu/common/Makefile
@@ -18,7 +18,6 @@
 
 obj-y	+= board-common.o
 obj-y	+= init.o
-obj-$(CONFIG_MVEBU_NAND)	+= nand_init.o
 obj-$(CONFIG_CMD_NET)		+= eth_init.o
 obj-$(CONFIG_MVEBU_SAR)		+= sar.o
 obj-$(CONFIG_DEVEL_BOARD)	+= devel-board.o
diff --git a/board/mvebu/common/nand_init.c b/board/mvebu/common/nand_init.c
deleted file mode 100644
index aefb37e..0000000
--- a/board/mvebu/common/nand_init.c
+++ /dev/null
@@ -1,27 +0,0 @@
-/*
- * ***************************************************************************
- * Copyright (C) 2015 Marvell International Ltd.
- * ***************************************************************************
- * This program is free software: you can redistribute it and/or modify it
- * under the terms of the GNU General Public License as published by the Free
- * Software Foundation, either version 2 of the License, or any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- * ***************************************************************************
- */
-
-#include <common.h>
-#include <nand.h>
-
-#include <asm/arch-mvebu/clock.h>
-#include <asm/arch-mvebu/driver_interface.h>
-
-void board_nand_init(void)
-{
-}
diff --git a/configs/mvebu_armada38x_defconfig b/configs/mvebu_armada38x_defconfig
index 50a6af1..d0b4525 100644
--- a/configs/mvebu_armada38x_defconfig
+++ b/configs/mvebu_armada38x_defconfig
@@ -1,11 +1,8 @@
 CONFIG_SPL=y
 +S:CONFIG_ARM=y
 +S:CONFIG_TARGET_ARMADA_38X=y
-CONFIG_MVEBU_NAND=y
-CONFIG_MVEBU_SPI=y
-CONFIG_MVEBU_PCI=y
++S:CONFIG_MVEBU_SPL_DIFFRENT_BASE_ADDR=y
 CONFIG_MVEBU_MPP_BUS=y
-+S:CONFIG_MVEBU_I2C=y
 +S:CONFIG_DEVEL_BOARD=y
 CONFIG_CUSTOMER_BOARD=y
 CONFIG_CMD_BDI=y
@@ -18,7 +15,6 @@ CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_LOADB=y
 CONFIG_CMD_LOADS=y
 CONFIG_CMD_FLASH=y
-CONFIG_CMD_I2C=y
 CONFIG_CMD_ECHO=y
 CONFIG_CMD_NET=y
 CONFIG_CMD_DHCP=y
@@ -33,9 +29,13 @@ CONFIG_CMD_MVEBU_TSEN=y
 +S:CONFIG_OF_CONTROL=y
 +S:CONFIG_OF_EMBED=y
 +S:CONFIG_DEFAULT_DEVICE_TREE="armada-385-db"
+CONFIG_MVEBU_PCI=y
+CONFIG_MVEBU_NAND=y
+CONFIG_NAND_PXA3XX=y
 CONFIG_MVEBU_MBUS=y
 +S:CONFIG_SYS_NS16550=y
++S:CONFIG_MVEBU_I2C=y
+CONFIG_MVEBU_SPI=y
 CONFIG_MVEBU_PINCTL=y
 +S:CONFIG_MVEBU_COMPHY_SUPPORT=y
 CONFIG_MVEBU_THERMAL_SENSOR=y
-+S:CONFIG_MVEBU_SPL_DIFFRENT_BASE_ADDR=y
diff --git a/drivers/mtd/nand/pxa3xx_nand.c b/drivers/mtd/nand/pxa3xx_nand.c
index 1565a9a..f3c6237 100644
--- a/drivers/mtd/nand/pxa3xx_nand.c
+++ b/drivers/mtd/nand/pxa3xx_nand.c
@@ -12,10 +12,10 @@
 #include <nand.h>
 #include <asm/errno.h>
 #include <asm/io.h>
-#include <asm/arch/cpu.h>
 #include <linux/mtd/mtd.h>
 #include <linux/mtd/nand.h>
 #include <linux/types.h>
+#include <asm/arch-mvebu/clock.h>
 
 #include "pxa3xx_nand.h"
 
diff --git a/include/configs/mvebu-common.h b/include/configs/mvebu-common.h
index e1ae55f..2e5a38f 100644
--- a/include/configs/mvebu-common.h
+++ b/include/configs/mvebu-common.h
@@ -394,9 +394,10 @@
 #ifdef CONFIG_MVEBU_NAND
 	#define CONFIG_SYS_MAX_NAND_DEVICE 1
 	#define CONFIG_SYS_NAND_MAX_CHIPS 1
-	#define CONFIG_SYS_NAND_SELF_INIT
 	#define CONFIG_NAND_ECC_STRENGTH 4
 	#define CONFIG_NAND_ECC_STEP_SIZE 512
+	#define CONFIG_SYS_NAND_ONFI_DETECTION
+	#define CONFIG_SYS_NAND_USE_FLASH_BBT
 #endif
 
 /* SPI Flash */
-- 
1.9.1

